SPECS/tdnf/tdnf.spec

```diff
@@ -121,23 +129,30 @@ systemd services for periodic automatic update
 %autosetup -p1
 
 %build
+%if 0%{?with_check}
+# remove problematic test file when running tests from within the rpm build directory
+rm pytests/tests/test_srpms.py
```

Comment:
Two questions here:
- What is wrong with the tests? I'm reluctant to remove tests without a good reason. If it's something on our end, we should fix it. If it's a different case, please add some details as to why it's OK to remove these tests.
- Is this really ran from the `%build` stage? It would be ideal, if we could keep all test steps inside the `%check` section, although I'm aware that's not always possible.

Reply:
The tests count and check the srpms present within the build expected RPM build directory. We don't necessarily know what exactly will be in this directory at the time of build, so these tests did not make much sense when this is the case and when being executed within the build directory. Additionally, the test's expected cleanup is to delete the entire directory. By doing so, the entire source directory and actively running tests are removed. Thus I've opted to remove thei file as to not be included prior to being built/included by cmake within the build directory.

Alternative to this, a skip directive could be added to each test, and the cleanup function could be removed as to no blow away the source. Though removing this in the spec `%build` or `%prep` section seemed much cleaner. I opted to include it within `%build` as it seemed more relevant to building rather than prepping to build.

Reply:
Got it. In that case, could I ask you to add that explanation you just gave here about why we need to remove the tests? A shorted version is fine, though.:)

And can you check, if moving this to the `%check` section is still fine? That way it's clearer this is a test-only change.

Reply:
I will check if its possible to remove during the check section, though I suspect it may have to remain in build or prep.

Reply:
Explanation added, also moved removal to check section. Since cmake duplicates the file in the build directory, it also is removed there for safety.